bin/bootstrap.sh

```diff
@@ -0,0 +1,202 @@
+if [ ! -f "${tfvars_file}.template" ]; then
+    echo "The file '${tfvars_file}.template' does not exist."
+fi
+if [ -f "$tfvars_file" ]; then
+    read -p "Are you sure you want to overwrite $tfvars_file? (y/n) " -n 1 -r
+    echo
+    if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+        echo "Aborting..."
+        exit 0
+    fi
+fi
+cat "${tfvars_file}.template" >"$tfvars_file"
+
+# Add a mandatory empty line at the end of the file
+echo "" >>$all_variables
+
+# Retrieving the names of variables declared in the file
+varnames=$(grep "^variable" $all_variables | sed 's/^.*variable "\(.*\)".*$/\1/p' | awk '!a[$0]++')
+
+# Creation of a temporary file that will contain the sorted and unique variables
+tmp_file=$(mktemp)
+while read -r varname; do
+    sed -n "/variable \"$varname\"/,/^}/p" $all_variables | sed -n '1,/^}/p' >>$tmp_file
+    echo "" >>$tmp_file
+done <<<"$varnames"
+
+# Delete duplicates
+cat $tmp_file >$all_variables
+# Delete the temporary file
+rm $tmp_file
+
+# Parse the $all_variables file and extract the variable names and descriptions
+mapfile -t variable_names < <(sed -n 's/^.*variable "\(.*\)".*$/\1/p' $all_variables)
+mapfile -t variable_descs < <(sed -n 's/^.*description *= *"\(.*\)".*$/\1/p' $all_variables)
+mapfile -t variable_types < <(sed -n 's/^.*type *= *\(.*\)$/\1/p' $all_variables)
+
+# Loop through each variable and prompt the user for a value
+for i in "${!variable_names[@]}"; do
+    var_name=${variable_names[i]}
+    var_desc=${variable_descs[i]}
+    var_type=${variable_types[i]}
+
+    # Extract the default value for the current variable (if available)
+    var_default=$(sed -n "/variable \"$var_name\"/,/^}/p" $all_variables | grep "default" | sed -E 's/^.*default *= *"?([^"]*)"?.*$/\1/')
+
+    # If the var_name doesn't exist in the .tfvars file, continue to the next loop iteration
+    if ! $(grep -q "$var_name" $tfvars_file); then
+        continue
+    fi
+
+    # Check if the variable has a default value
+    if [ -n "$var_default" ]; then
+        default_variables+=("$var_name")
+        default_descs+=("$var_desc")
+        default_defaults+=("$var_default")
+        default_types+=("$var_type")
+    else
+        non_default_variables+=("$var_name")
+        non_default_descs+=("$var_desc")
+        non_default_types+=("$var_type")
+    fi
+done
+
+# Loop through variables with no default values first
+for i in "${!non_default_variables[@]}"; do
+    var_name=${non_default_variables[i]}
+    var_desc=${non_default_descs[i]}
+    var_type=${non_default_types[i]}
+
+    if [ "$var_name" == "argocd_password" ]; then
+        read -p "ArgoCD password : " argocd_password
+        while [ -z "$argocd_password" ]; do
+            read -p "You have to specify a value for \"argocd_password\": " argocd_password
+        done
+        DOCKER_USER="$(id -u):$(id -g)" \
+        argocd_password_hashed="$(docker-compose run --quiet-pull --rm argocd-cli argocd account bcrypt --password $argocd_password)"
```

Comment:
```suggestion
        argocd_password_hashed="$(docker-compose run --rm argocd-cli argocd account bcrypt --password $argocd_password)"
```

Sous Linux l'option `--quiet-pull` fonctionne uniquement avec `docker compose` mais pas avec `docker-compose`... C'est p'être un conflit de mon côté